src/core/server/logging/integration_tests/rolling_file_appender.test.ts

```diff
@@ -0,0 +1,220 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { join } from 'path';
+import { rmdirSync, mkdtempSync, readFileSync, readdirSync } from 'fs';
+import moment from 'moment-timezone';
+import * as kbnTestServer from '../../../test_helpers/kbn_server';
+import { getNextRollingTime } from '../appenders/rolling_file/policies/time_interval/get_next_rolling_time';
+
+const flushDelay = 250;
+const delay = (waitInMs: number) => new Promise((resolve) => setTimeout(resolve, waitInMs));
+const flush = async () => delay(flushDelay);
+
+function createRoot(appenderConfig: any) {
+  return kbnTestServer.createRoot({
+    logging: {
+      silent: true, // set "true" in kbnTestServer
+      appenders: {
+        'rolling-file': appenderConfig,
+      },
+      loggers: [
+        {
+          context: 'test.rolling.file',
+          appenders: ['rolling-file'],
+          level: 'debug',
+        },
+      ],
+    },
+  });
+}
+
+describe('RollingFileAppender', () => {
+  let root: ReturnType<typeof createRoot>;
+  let testDir: string;
+  let logFile: string;
+
+  const getFileContent = (basename: string) =>
+    readFileSync(join(testDir, basename)).toString('utf-8');
+
+  beforeEach(async () => {
+    testDir = mkdtempSync('rolling-test');
```

Comment:
why not async counterparts?

Reply:
No reasons, I guess it was just late when I did it. Will change